github-actions/runtime-variables/RuntimeVariables.MD

```diff
@@ -0,0 +1,234 @@
+
+```
+  run: |
+    if ${COMMIT_VAR} == true; then
+      echo "::set-env name=flag::true"
+      echo "flag set to true"
+    else
+      echo "::set-env name=flag::false"
+      echo "flag set to false"
+    fi
+```
+
+The `run` command here checks to see if the `COMMIT_VAR` variable has been set to `true`, and if it has, it sets a secondary flag to true, and echoes this behavior. It does the same if the variable is `false`.
+
+The specific reason to do this is to allow for the `flag` variable to be used in further steps instead of having to reuse the `COMMIT_VAR` in every step. Further, it allows for the flag to be used in the `if` step of an action, as in the next part of the snippet.
+
+```
+- name: "Use flag if true"
+  if: env.flag
+  run: echo "Flag is available and true"
+```
+
+In this part of the snippet, the next step in the same job is now using the `flag` that was set in the previous step. This allows the user to:
+
+1. Reuse the flag instead of repeatedly accessing the Github Context
+2. Set the flag using multiple conditions, instead of just one. For example, a different step might ALSO set the flag to `true` or `false` for different reasons. 
+3. Change the variable in exactly one place instead of having to change it in multiple places
+
+#### Shorter Alternative
+
+The "Set flag from commit" step can be simplified to the following in order to make the code much shorter, although not necessarily more readable:
+
+```
+- name: "Set flag from Commit"
+  env:
+    COMMIT_VAR: ${{ contains(github.event.head_commit.message, '[commit var]') }}
+  run: |
+    echo "::set-env name=flag::${COMMIT_VAR}"
+    echo "set flag to ${COMMIT_VAR}"
+```
+
+#### Usage
+
+##### Including the Variable
+
+1. Push to branch `master`:
+
+```
+> git add.
+> git commit -m "Running Github Actions Test [commit var]"
+> git push
+```
+
+2. This triggers the workflow (as will any push). As the `[commit var]` is in the commit message, the `${COMMIT_VAR}` variable in the workflow will be set to `true` and result in the following:
+
+![Commit True Scenario](images/CommitTrue.PNG)
+
+##### Not Including the Variable
+
+
+1. Push to  branch `master`:
+
+```
+> git add.
+> git commit -m "Running Github Actions Test"
+> git push
+```
+
+2. This triggers the workflow (as will any push). As the `[commit var]` is **not** in the commit message, the `${COMMIT_VAR}` variable in the workflow will be set to `false` and result in the following:
+
+![Commit True Scenario](images/CommitFalse.PNG)
+
+
+### PR Body Variables
+
+When a PR is made, the PR Body can also be used to setup variables. These variables can be made available to all the workflow runs that stem from that PR, which can help ensure that commit messages are more informative and less cluttered, and reduces the work on the developer.
+
+Once again, this for an expected key and value. In this case, the key is `PR_VAR` and the value is `[pr var]`.
```

Comment:
It might be worth noting that with PR templates, you can "systematize" this idea a bit more across a team

Reply:
Done! I did this in the objective after adding an `in scope` and `out of scope` section, with the out of scope serving as a place for "ideas that someone else should do but I don't think belong in this guide".